rfcs/0013-deprecate-currency-factory.md

```diff
@@ -0,0 +1,345 @@
+these asset classifications. Another pallet, pallet-asset-manager, is used to 
+route between the two.
+
+Local asset IDs are determined by the result of hashing a counter that tracks
+the number of local assets. While Foreign asset IDs are determined by the assets
+XCM Multilocation.
+
+#### Foreign
+
+Local assets are registered with a `ForeignAssetType`, `AssetRegisterarMetadata`, a `Balance` representing the ED, and a `bool` `is_sufficent`.
+
+## Requirements
+
+If we are to deprecate Currency Factory, we will need to ensure the requirements
+it fulfills are still met by some solution.
+
+* Permissioned asset creation can be done without collision
+
+* Asset metadata is available
+
+* Can registrar both local and foreign asset types
+
+* Support XCM Multi Locations for foreign assets
+
+## Solution Proposals
+
+There are multiple ways we could go about replacing Currency Factory. Possible 
+solutions are not necessarily mutually exclusive (i.e. we could implement multiple). 
+These solutions are detailed below:
+
+### Simplify & Move the Asset ID Reservation System
+
+One way to remove the need for pallet-currency-factory is to move its 
+functionality to pallet-asset-registry. If we also simplify the asset ID 
+reservation system to no longer use predefined ranges, this will be a minimal 
+change.
+
+This retains the functionality of Currency Factory, and does not require a new 
+pallet. However, this does not enable permissionless asset creation without the
+need for more design.
+
+#### Technical Implementation
+
+Instead of reserving asset IDs via our current range system, we could simply 
+have one nonce for reserving new asset IDs, which would (by the definition of a 
+nonce) ensure no collisions. This nonce can start at an arbitrary but high 
+number so that our hard-coded asset IDs are still safe. This would reduce the 
+complexity of our current reservation system while still avoiding collision.
+
+#### Consequences
+
+* If we already have assets in various ranges, the runtime migration process
+will become more complicated as we translate them to existing in one range.
+  
+### Remove Asset ID Reservation
+
+> We do not need [asset ID reservation]. We just runtime config assets to be 
+permissioned. When we will have permisionless assets they may already have 
+reservation (any way making fork with reservation is easy later).
+
+Asset reservation helps us automate new asset IDs. However, for the scope of 
+release two - this may not be necessary. We could instead hard-code asset IDs 
+for LP tokens and other instances.
+
+Given we can currently manage without automated asset creation, as we can forgo
+the asset creation requirements for release two and ensure that all asset data
+is still made available.
+
+#### Consequences
+
+* Minor runtime migration will be needed for deleting Currency Factory's 
+storage.
+
+### Use an Externally Maintained Asset ID Reservation System
```

Comment:
This is a separate concern as proposed before, we can focus on concern one for release 2. But not a problem to document the solution. I agree in  concept though we need to assess how well it supports our requirements such as asset locking at the time of implementation(release 3 or after)